ssh/ssh_test.go

```diff
@@ -0,0 +1,167 @@
+package ssh
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
```

Comment:
Are we going to remove the SSH functionality from Terratest? If so, will we get a circular dependency later on if this code depends on Terratest, and Terratest, in turn, depends on this code?

Reply:
> Will we remove this functionality from Terratest? Or update the Terratest functions to call this library under the hood?

I think eventually we will update the terratest functions to call this library, although there are a few kinks to work out before we can do that (namely adding support for SCP and figuring out logging). Probably not worth the effort in the short term.

> If so, will we get a circular dependency later on if this code depends on Terratest, and Terratest, in turn, depends on this code?

I'll have to do some research on this, but in theory I think we can avoid the circular dependency because these terratest packages don't pull in the `ssh` module in `terratest`.

Note that `terratest` already uses the `errors` package in `gruntwork-cli` in a few places, and go didn't complain, so I think what I said above works.

> Do we need to update any docs in this repo?

Ah yup good catch. Will make a commit to update.

Reply:
Ok added package docs to root README: c38d9dd

NOTE: there appears to be some weird go module linking issue, where it wants to link in `gruntwork-cli` at `v0.7.0` in `go.sum`, but not sure why. `go mod tidy` won't clean it up despite there being 0 references to `gruntwork-cli` in this repo. I think we'll have to leave it in for now.

Reply:
One random thought on `go mod tidy`: did you rename your local checkout from `gruntwork-cli` to `go-commons`? And update `git remote` to `go-commons`? I had issues with Go dependencies not resolving correct in `terraform-aws-security` because my local checkout was still sitting in a `module-security` folder. Only after renaming and updating `git remote` (not sure which mattered!) did dependencies resolve correctly.

Reply:
Yup I did that. I actually figured it out. It's coming from `terratest`, which depends on `gruntwork-cli` still.

Reply:
Ah, got it